### PR TITLE
fix: Consider URL encoded recordings

### LINKED ISSUE #24: [BUG] Hiding of sensitive information doesn't work within URL encoded recordings
## Expected Behavior
Hide sensitive information even when they're URL encoded.

## Current Behavior
Does not find sensitive information in URL encoded recording and therefore it can't be hidden.

## Possible Solution
Hide them with custom [processing functions](https://github.com/superfaceai/testing#recording).

## Steps to Reproduce
https://github.com/superfaceai/station/blob/main/grid/oauth2/refresh-token/maps/refresh-token.ts

### PR DESCRIPTION
**Pull request by martinalbert**
## Description

Fixes #24 

## Motivation and Context

URL encoded contents of recordings don't get replaced properly

## Types of changes

- [x] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to change)

## Checklist:

- [ ] I have updated the documentation accordingly. If you are changing **code related to user secrets** you need to really make sure that [security documentation](SECURITY.md) is correct.
- [x] I have read the **CONTRIBUTION_GUIDE** document.
- [x] I haven't repeated the code. (DRY)
- [x] I have added tests to cover my changes.
- [x] All new and existing tests passed.

### CODE REVIEW

File: src/nock.utils.test.ts [hunk 1]
```diff
@@ -471,16 +472,151 @@ describe('nock utils', () => {
...
+          path: '/get?text=123',
+          method: 'GET',
+          status: 200,
+          body: {
+            my_api_key: TMP_PLACEHOLDER,
+          },
+        });
+      });
+
+      it('replaces placeholder for api key in body', async () => {
```
Comment (janhalama) [inline src/nock.utils.test.ts]: Do I understand it right that `replaceCredentialInDefinition` is used also to replace place holder with credential when running tests (without recording)?

**Comment by janhalama** [comment]
I would consider splitting /src/nock.utils.ts. The test file has > 1200 lines.

File: src/nock.utils.test.ts [hunk 2]
```diff
@@ -698,16 +834,127 @@ describe('nock utils', () => {
         });
       });
     });
+
+    describe('when sensitive information are URL encoded', () => {
+      it('replaces api key in body', async () => {
```
Comment (janhalama) [inline src/nock.utils.test.ts]:
```suggestion
      it('replaces parameter in body', async () => {
```

Here we are in integration test section.

File: src/nock.utils.test.ts [hunk 3]
```diff
@@ -698,16 +834,127 @@ describe('nock utils', () => {
...
+          path: '/get?text=123',
+          method: 'GET',
+          status: 200,
+          body: {
+            my_api_key: TMP_PLACEHOLDER,
+          },
+        });
+      });
+
+      it('replaces placeholder for api key in body', async () => {
```
Comment (janhalama) [inline src/nock.utils.test.ts]:
```suggestion
      it('replaces placeholder for parameter in body', async () => {
```

File: src/nock.utils.test.ts [hunk 4]
```diff
@@ -698,16 +834,127 @@ describe('nock utils', () => {
...
+          path: '/get?text=123',
+          method: 'GET',
+          status: 200,
+          body: {
+            my_api_key: secret,
+          },
+        });
+      });
+
+      it('replaces apikey in query', () => {
```
Comment (janhalama) [inline src/nock.utils.test.ts]:
```suggestion
      it('replaces parameter in query', () => {
```

File: src/nock.utils.test.ts [hunk 5]
```diff
@@ -698,16 +834,127 @@ describe('nock utils', () => {
...
+
+        expect(definition).toEqual({
+          scope: BASE_URL,
+          path: `/get?api_key=${TMP_PLACEHOLDER}&text=123`,
+          method: 'GET',
+          status: 200,
+        });
+      });
+
+      it('replaces apikey in path', () => {
```
Comment (janhalama) [inline src/nock.utils.test.ts]:
```suggestion
      it('replaces parameter in path', () => {
```

File: src/nock.utils.test.ts [hunk 6]
```diff
@@ -900,5 +1147,118 @@ describe('nock utils', () => {
         });
       });
     });
+
+    describe('when sensitive information are URL encoded', () => {
+      it('replaces api key in body', async () => {
```
Comment (janhalama) [inline src/nock.utils.test.ts]:
```suggestion
      it('replaces input in body', async () => {
```

File: src/nock.utils.test.ts [hunk 7]
```diff
@@ -900,5 +1147,118 @@ describe('nock utils', () => {
...
+          path: '/get?text=123',
+          method: 'GET',
+          status: 200,
+          body: {
+            my_api_key: TMP_PLACEHOLDER,
+          },
+        });
+      });
+
+      it('replaces placeholder for api key in body', async () => {
```
Comment (janhalama) [inline src/nock.utils.test.ts]:
```suggestion
      it('replaces placeholder for input in body', async () => {
```

File: src/nock.utils.test.ts [hunk 8]
```diff
@@ -900,5 +1147,118 @@ describe('nock utils', () => {
...
+          path: '/get?text=123',
+          method: 'GET',
+          status: 200,
+          body: {
+            my_api_key: secret,
+          },
+        });
+      });
+
+      it('replaces apikey in query', () => {
```
Comment (janhalama) [inline src/nock.utils.test.ts]:
```suggestion
      it('replaces input in query', () => {
```

File: src/nock.utils.test.ts [hunk 9]
```diff
@@ -900,5 +1147,118 @@ describe('nock utils', () => {
...
+
+        expect(definition).toEqual({
+          scope: BASE_URL,
+          path: `/get?api_key=${TMP_PLACEHOLDER}&text=123`,
+          method: 'GET',
+          status: 200,
+        });
+      });
+
+      it('replaces apikey in path', () => {
```
Comment (janhalama) [inline src/nock.utils.test.ts]:
```suggestion
      it('replaces input in path', () => {
```

**Review by janhalama — COMMENTED** [review]
I found some minor copy-paste issues in tests, otherwise looks good.

File: src/nock.utils.test.ts (hunk 1 above)
Comment (martinalbert) [reply src/nock.utils.test.ts]: Yes, should I find better naming for this?

Comment (janhalama) [reply src/nock.utils.test.ts]: No I just was not sure why the test is here.

**Review by janhalama — APPROVED** [review]
LGTM 👍

**Pull request merged**